### PR TITLE
feat: configurable sysctl.d configuration

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR will allow user-configurable sysctl.d Linux kernel runtime configuration.

No kernel configuration changes are landing here; all pre-existing aks-engine kernel params tuning has been standardized into the new `sysctldConfig` interface:

```
"kubernetesConfig": {
    "sysctldConfig": {
        "net.ipv4.tcp_keepalive_time": "120",
        "net.ipv4.tcp_keepalive_intvl": "75",
        "net.ipv4.tcp_keepalive_probes": "9"
    }
}
```

(The above is an example of passing custom config.)

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_config.sh [hunk 1]
```diff
@@ -264,7 +264,6 @@ configureCNIIPTables() {
 {{if NeedsContainerd}}
 ensureContainerd() {
     wait_for_file 1200 1 /etc/systemd/system/containerd.service.d/exec_start.conf || exit $ERR_FILE_WATCH_TIMEOUT
-    wait_for_file 1200 1 /etc/sysctl.d/11-containerd.conf || exit $ERR_FILE_WATCH_TIMEOUT
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/cse_config.sh]: Moved into the new, general `11-aks-engine.conf` sysctl file

File: test/e2e/kubernetes/kubernetes_test.go [hunk 2]
```diff
@@ -336,6 +336,35 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 			}
 		})
 
+		It("should validate aks-engine-provided sysctl configuration", func() {
```
Comment (jackfrancis) [inline test/e2e/kubernetes/kubernetes_test.go]: This is rather manual, but it's proof that our new sysctldConfig interface is doing exactly what we think it's doing.

File: parts/k8s/cloud-init/artifacts/kubelet.service [hunk 3]
```diff
@@ -16,15 +16,6 @@ ExecStartPre=/bin/bash -c "if [ $(mount | grep \"/var/lib/kubelet\" | wc -l) -le
 ExecStartPre=/bin/mount --make-shared /var/lib/kubelet
 {{/* This is a partial workaround to this upstream Kubernetes issue: */}}
 {{/* https://github.com/kubernetes/kubernetes/issues/41916#issuecomment-312428731 */}}
-ExecStartPre=/sbin/sysctl -w net.ipv4.tcp_retries2=8
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/kubelet.service]: @xizhamsft @yangl900 @xuto2 FYI, we are moving these sysctl configs into the new api model interface `sysctldConfig` (available for masterProfile and agentPoolProfiles).

No kernel parameter configuration changes, just changing how they are implemented in aks-engine.

These aks-engine changes will allow for rapid experimentation of various kernel params, no need to change code.

Comment (xuto2) [reply parts/k8s/cloud-init/artifacts/kubelet.service]: @jackfrancis  Good to know. Thanks for the notice!

**Comment by sjwoodard** [comment]
Is this available on AKS? I'm not clear on how we can we pass in kernel params?

**Comment by jackfrancis** [comment]
Hi @sjwoodard, I don't think so. @xuto2 is there any AKS config interface to allow users to customize sysctl.d settings on AKS nodes?

**Comment by xuto2** [comment]
not yet but AKS is working on that, plan to have it ready in 2020-11-01 API version.

**Pull request merged**